src/Features/Core/Portable/CodeRefactorings/UseNamedArguments/AbstractUseNamedArgumentsCodeRefactoringProvider.cs

```diff
@@ -95,28 +94,36 @@ protected abstract class Analyzer<TBaseArgumentSyntax, TArgumentSyntax, TArgumen
                     return;
                 }
 
-                var argumentList = (TArgumentListSyntax)argument.Parent;
-                var (index, count) = GetArgumentListIndexAndCount(argument, argumentList);
+                var argumentList = argument.Parent as TArgumentListSyntax;
+                if (argumentList == null)
+                {
+                    return;
+                }
 
                 var arguments = GetArguments(argumentList);
-                for (var i = index; i < count; i++)
+                var argumentCount = arguments.Count;
+                var argumentIndex = arguments.IndexOf(argument);
+                if (argumentIndex >= parameters.Length)
                 {
-                    if (!(arguments[i] is TArgumentSyntax))
-                    {
-                        return;
-                    }
+                    return;
                 }
 
-                if (!IsLegalToAddNamedArguments(parameters, count))
+                if (!IsLegalToAddNamedArguments(parameters, argumentCount))
                 {
                     return;
                 }
 
-                var argumentName = parameters[index].Name;
-                context.RegisterRefactoring(
-                    new MyCodeAction(
-                        string.Format(FeaturesResources.Add_argument_name_0, argumentName),
-                        c => AddNamedArgumentsAsync(root, document, argument, parameters, index)));
+                for (var i = argumentIndex; i < argumentCount; i++)
+                {
+                    if (!(arguments[i] is TArgumentSyntax))
```

Comment:
Ah, i think we need to call this TSimpleArgumentSyntax then to better convey how it will be instantiated and what it is expected to match against.